seed/static/seed/js/controllers/inventory_list_controller.js

```diff
@@ -111,6 +118,201 @@ angular.module('BE.seed.controller.inventory_list', [])
+
+        var modalInstance = $uibModal.open({
+          templateUrl: urls.static_url + 'seed/partials/filter_group_modal.html',
+          controller: 'filter_group_modal_controller',
+          resolve: {
+            action: _.constant('rename'),
+            data: _.constant($scope.currentFilterGroup),
+          }
+        });
+
+        modalInstance.result.then(function (newName) {
+          $scope.currentFilterGroup.name = newName;
+          _.find($scope.filterGroups, {id: $scope.currentFilterGroup.id}).name = newName;
+          Notification.primary('Renamed ' + oldFilterGroup.name + ' to ' + newName);
+        });
+      };
+
+      $scope.save_filter_group = function () {
+        var label_ids = [];
+        for (label in $scope.selected_labels) {
+            label_ids.push($scope.selected_labels[label].id);
+        };
+        filter_group_inventory_type = $scope.inventory_type === 'properties' ? 'Property' : 'Tax Lot';
+        var query_dict = inventory_service.get_format_column_filters($scope.column_filters);
+        var filterGroupData = {
+          "name": $scope.currentFilterGroup.name,
+          "query_dict": query_dict,
+          "inventory_type": filter_group_inventory_type,
+          "labels": label_ids,
+          "label_logic": $scope.labelLogic
+        };
+        filter_groups_service.update_filter_group($scope.currentFilterGroup.id, filterGroupData).then(function (result) {
+          $scope.currentFilterGroup = result;
+          $scope.Modified=false;
+          Notification.primary('Saved ' + $scope.currentFilterGroup.name);
+        });
+      };
+
+
+      // compare filters if different then true, then compare labels, and finally label_logic. All must be the same to return false
+      $scope.isModified = function () {
+        current_filters = {};
+        current_filters = inventory_service.get_format_column_filters($scope.column_filters);
+        saved_filters = $scope.currentFilterGroup.query_dict;
+        current_labels = [];
+        for (label in $scope.selected_labels) {
+          current_labels.push($scope.selected_labels[label].id);
+        };
+        saved_labels = $scope.currentFilterGroup.labels;
+        current_label_logic = $scope.labelLogic;
+        saved_label_logic = $scope.currentFilterGroup.label_logic;
+        if (!_.isEqual(current_filters, saved_filters)) {
+          $scope.Modified = true
+          console.log("filters are different");
+        } else if (!_.isEqual(current_labels.sort(), saved_labels.sort())) {
+          $scope.Modified = true
+          console.log("labels are different");
+        } else if (current_label_logic !== saved_label_logic) {
+          $scope.Modified = true
+          console.log("label_logic is different");
+        } else {
+          $scope.Modified = false
+          console.log("no changes");
+        }
+        return $scope.Modified;
+      };
+
+      $scope.check_for_changes = function () {
```

Comment:
This could have a more descriptive name. 
```suggestion
      $scope.check_for_filter_group_changes = function () {
```

Reply:
Thank you for the suggestion.